x/conflict/keeper/conflict.go

```diff
@@ -134,29 +153,173 @@ func (k Keeper) ValidateResponseConflict(ctx sdk.Context, conflictData *types.Re
 			return fmt.Errorf("AccAddressFromHex %s provider ResponseFinalizationData: %w", print_st, err)
 		}
 		if !derived_providerAccAddress.Equals(expectedAddress) {
-			return fmt.Errorf("mismatching %s provider address signature and responseFinazalizationData %s , %s", print_st, derived_providerAccAddress, expectedAddress)
+			return fmt.Errorf("mismatching %s provider address signature and responseFinalizationData %s , %s", print_st, derived_providerAccAddress, expectedAddress)
 		}
 		// validate the responses are finalized
-		if !k.specKeeper.IsFinalizedBlock(ctx, chainID, request.RelayData.RequestBlock, response.LatestBlock) {
-			return fmt.Errorf("block isn't finalized on %s provider! %d,%d ", print_st, request.RelayData.RequestBlock, response.LatestBlock)
+		if !k.specKeeper.IsFinalizedBlock(ctx, chainID, request.RelayData.RequestBlock, replyMetadata.LatestBlock) {
+			return fmt.Errorf("block isn't finalized on %s provider! %d,%d ", print_st, request.RelayData.RequestBlock, replyMetadata.LatestBlock)
 		}
 		return nil
 	}
+
 	err = validateResponseFinalizationData(providerAccAddress0, conflictData.ConflictRelayData0.Reply, conflictData.ConflictRelayData0.Request, true)
 	if err != nil {
 		return err
 	}
+
 	err = validateResponseFinalizationData(providerAccAddress1, conflictData.ConflictRelayData1.Reply, conflictData.ConflictRelayData1.Request, true)
 	if err != nil {
 		return err
 	}
+
 	// 5. validate mismatching responses
 	if bytes.Equal(conflictData.ConflictRelayData0.Reply.HashAllDataHash, conflictData.ConflictRelayData1.Reply.HashAllDataHash) {
 		return fmt.Errorf("no conflict between providers data responses, its the same")
 	}
+
 	return nil
 }
 
-func (k Keeper) ValidateSameProviderConflict(ctx sdk.Context, conflictData *types.FinalizationConflict, clientAddr sdk.AccAddress) error {
+func (k Keeper) ValidateSameProviderConflict(ctx sdk.Context, conflictData *types.FinalizationConflict, clientAddr sdk.AccAddress) (mismatchingBlockHeight int64, mismatchingBlockHashes map[string]string, err error) {
+	// TODO: Validate BlockDistanceFromFinalization with spec
```

Comment:
In `validateFinalizedBlock` I just take this number from the spec.
Since we don't send this number in the proto anymore, this TODO is not irrelevant.